mockall_derive/src/mock_function.rs

```diff
@@ -1211,7 +1211,7 @@ impl<'a> ToTokens for CommonExpectationsMethods<'a> {
                 {
                     self.0.push(Expectation::default());
                     let __mockall_l = self.0.len();
-                    &mut self.0[__mockall_l - 1]
+                    self.0.get_mut(__mockall_l - 1).unwrap()
```

Comment:
Isn't a `get_mut` followed by an `unwrap` equivalent to slicing?  It seems like it would just change one panic message for another.  A change that would at least improve readability would be to use `last_mut` instead.  That would be usable here and in the other locations too.
```suggestion
                    self.0.last_mut().unwrap()
```

Reply:
>  It seems like it would just change one panic message for another.

You are correct, I use this lint to avoid implicit panics in code. With the `mockall` library a panic is desirable as it should only be run during tests. If you'd prefer not to have the additional `.unwrap()`'s added I'm happy to change this to add `#[allow(clippy::indexing_slicing)]` tags where appropriate, or change to `.expect()` calls.

The goal is to allow the proc-macro to compile for end-users with `#[deny(clippy::indexing_slicing)]` set on their projects, as right now it generates an error from inside the proc macro.

Reply:
I don't think that replacing slicing with .get_mut offers any advantages to Mockall, so I don't want to do it.  But using last_mut() instead of slicing removes a local variable, and that's a good thing.  BTW, you can also use last_mut() in all of the other places that this PR affects; not just the one I pointed out.

Reply:
I understand the unwillingness to update the library - this doesn't change any functionality.

Do you have a recommendation on how better to address this issue? Not sure on how best to handle errors from inside a proc macro like this.

![image](https://github.com/asomers/mockall/assets/29187591/47fa82df-e6cd-42bd-8729-4ae0ca59429e)

Reply:
You are right, this does not offer material changes to mockall. I'm unsure how other to resolve the error from inside the proc macro though: [https://pastebin.com/K8XTbuJ6](https://pastebin.com/K8XTbuJ6)

Do you have advice on how I can best resolve this in my crate?

Because there are multiple methods annotating with `#[allow()]` does not work. :(

![image](https://github.com/asomers/mockall/assets/29187591/010776fd-a264-41d3-be98-8efa0c1151c4)

Reply:
It would be fine to add an `#[allow()]` to the generated code.  There are already several such allowances.